pkg/linux/entropy.go

```diff
@@ -0,0 +1,104 @@
+package linux
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+)
+
+var (
+	trngList map[string]int
+)
+
+const (
+	hwRandomDir           string        = "/sys/class/misc/hw_random"
+	hwRandomCurrentFile   string        = "/sys/class/misc/hw_random/rng_current"
+	hwRandomAvailableFile string        = "/sys/class/misc/hw_random/rng_available"
+	entropyChunkSize      int64         = 128
+	entropyFeedTime       time.Duration = 2
+	rebootTimeout         time.Duration = 5
+	randomDevice          string        = "/dev/random"
+	hwRandomDevice        string        = "/dev/hwrng"
+)
+
+func init() {
+	trngList = make(map[string]int)
+
+	// Priorize the HW RNG 0 is the highest
+	trngList["tpm-rng"] = 0
```

Comment:
Aren't all of these except tpm-rng already used by the kernel inside of /dev/random ?

Reply:
Not that I am aware of it. See rng-tools

Reply:
After a quick look at `drivers/char/hw_random`, it seems you're right: the kernel would start a hwrng thread filling /dev/random when an rng is set as rng_current if the quality is >0, but none of these drivers set their quality, and there isn't an userspace interface to set quality / force the hwrng kthread to start anyway.
ChromeOS even has a patch to force the use of the tpm and assign it maximum quality (https://groups.google.com/a/chromium.org/forum/#!msg/chromium-os-checkins/Leyl1e-WC5k/fRImmEXvBAAJ) I wonder if we could carry a similar one